hands-on-exercise/1-native/3-add-game.md

```diff
@@ -0,0 +1,314 @@
+---
+title: "Add your first object"
+order: 4
+description: Where you make the checkers module interesting
+tags:
+  - guided-coding
+  - cosmos-sdk
+---
+
+# Add your first object
+
+After the [previous section](./2-build-module.md), you have a somewhat empty checkers module integrated in a minimal app. It is now time to:
+
+* Add a game storage type.
+* Compile Protobuf.
+* Add the necessary keeper functions.
+
+## The game rules
+
+A good start to developing a checkers blockchain is to define the rule set of the game. There are many versions of the rules. Choose [a very simple set of basic rules](https://www.ducksters.com/games/checkers_rules.php) to avoid getting lost in the rules of checkers or the proper implementation of the board state.
+
+Use [a ready-made implementation](https://github.com/batkinson/checkers-go/blob/a09daeb/checkers/checkers.go) with the additional rule that the board is 8x8, is played on black cells, and black plays first. This code will not need adjustments. Copy this rules file into a `rules` folder inside your module. Change its package from `checkers` to `rules`. You can do this by command-line:
```

Comment:
> Copy this rules file into a `rules` folder inside your module.

Specify the whole path. Unclear where the `rules` folder should go, e.g., `checkers-minimal` or `checkers-minimal/module`?

Reply:
Adding that.